test/unit/slave/test_cluster_slave.py

```diff
@@ -182,7 +182,7 @@ def test_setup_should_send_correct_state_update_to_master(self, is_setup_success
         slave.connect_to_master(self._FAKE_MASTER_URL)
         slave._project_type = MagicMock()
         if not is_setup_successful:
-            slave._project_type.setup_build.side_effect = SetupFailureError
+            slave._project_type.fetch_project.side_effect = SetupFailureError
 
         slave._async_setup_build(executors=[], project_type_params={})
 
```

Comment:
Can you add an assert for the `run_job_config_setup()` call? I'm not sure if this is the test that it would go in but it seems like it should go somewhere. :)